src/rest-api.js

```diff
@@ -47,8 +52,29 @@ function perform(config = {}, method = "get", path, params = {}) {
 
     query
     .on("success", actions.resolve)
-    .on("error", actions.reject)
-    .on("fail", actions.reject);
+    .on("error", actions.reject);
+
+    if (method === "get") {
```

Comment:
@michaloo I think it's weird and a little misleading to set `options.timeout = 10000` here. This value is not used anywhere as far as I can see...

Let set it above before the request is actually sent and fallback 10k if there is no value set on `get` requests... And more generally, we should pretty much never have infinite timeouts since the server side API does timeout at 30s max so there should be a default value for timeouts all the time in the request options